portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLAppServiceImpl.java

```diff
@@ -3506,6 +3527,20 @@ protected Folder moveFolder(
 	@BeanReference(type = RepositoryProvider.class)
 	protected RepositoryProvider repositoryProvider;
 
+	private void _copyResourcePermissions(
+			long companyId, String className, long sourceResourcePrimKey,
+			long targetResourcePrimKey)
+		throws PortalException {
+
+		for (int scope : ResourceConstants.SCOPES) {
+			_resourcePermissionLocalService.deleteResourcePermissions(
+				companyId, className, scope, targetResourcePrimKey);
+		}
+
+		_resourceLocalService.copyModelResources(
```

Comment:
have a question here.

Only a security issue (that can make this task harder)
i'm little concerned that the site permissions between the sites/asset libs, that are different can be transferred without the knowledge of the user.

- See the video with the behavior

https://github.com/liferay-lima/liferay-portal/assets/47524751/408a45c3-42df-4f17-a716-ac14f40c7c38



maybe I'm reading more that actually is.

but I prefer to make it notice to you

Other way, the PR work as a charm and LGTM